src/membership/view.rs

```diff
@@ -1,15 +1,295 @@
+use crate::{
+    common::{ConfigId, Endpoint, NodeId},
+    error::{Error, Result},
+};
+use std::collections::HashSet;
+
+type Ring = crate::membership::ring::Ring<Endpoint>;
+
 #[derive(Debug, Clone)]
 pub struct View {
-    configuration: Configuration,
+    k: i32,
+    rings: Vec<Ring>,
```

Comment:
One day when rust gets specialization, we can make this an array of size `k`...one day...